framework/src/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1257,7 +1259,7 @@ private boolean checkSuppressWarnings(String /*@Nullable*/ [] userSwKeys, String
      *         it is contained by a declaration with an appropriately-valued
      *         {@literal @}SuppressWarnings annotation; false otherwise
      */
-    private boolean shouldSuppressWarnings(Tree tree, String errKey) {
+    public boolean shouldSuppressWarnings(Tree tree, String errKey) {
```

Comment:
You shouldn't make this method private. In your usecase you always pass null as `errKey`. Document a dedicated method for this. Maybe it can be a lot simpler without errKey.